src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs

```diff
@@ -1,148 +1,351 @@
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_before_semicolon_in_for_statement"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceBeforeSemicolonsInForStatement")});
 
         public static Option<BinaryOperatorSpacingOptions> SpacingAroundBinaryOperator { get; } = new Option<BinaryOperatorSpacingOptions>(nameof(CSharpFormattingOptions), nameof(SpacingAroundBinaryOperator), defaultValue: BinaryOperatorSpacingOptions.Single,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpacingAroundBinaryOperator"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_around_operators", ParseEditorConfigSpacingAroundBinaryOperator),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpacingAroundBinaryOperator")});
+
+        private static object ParseEditorConfigSpacingAroundBinaryOperator(string binaryOperatorSpacingValue)
+        {
+            switch (binaryOperatorSpacingValue)
+            {
+                case "ignore": return BinaryOperatorSpacingOptions.Ignore;
+                case "none": return BinaryOperatorSpacingOptions.Remove;
+                case "before_and_after": return BinaryOperatorSpacingOptions.Single;
+                default: return BinaryOperatorSpacingOptions.Single;
+            }
+        }
 
         public static Option<bool> IndentBraces { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentBraces), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.OpenCloseBracesIndent"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_braces"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.OpenCloseBracesIndent")});
 
         public static Option<bool> IndentBlock { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentBlock), defaultValue: true,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentBlock"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_block_contents"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentBlock")});
 
         public static Option<bool> IndentSwitchSection { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(IndentSwitchSection), defaultValue: true,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.IndentSwitchSection"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_indent_case_labels"),
```

Comment:
it also indents `default:` which is probably why they call it "SwitchSection" in the option name. Not sure I care, though.